tools/constants/constants.go

```diff
@@ -24,3 +24,20 @@ const (
 	// Components is the placeholder for the resource type of components
 	Components ResourceType = "Components"
 )
+
+// DefaultFloat32Value is the exported constant value for the default float32 value for an uninitialized float32
+// variable.
+const (
+	DefaultFloat32Value float32 = iota
+)
+
+const (
+	// VersionNotNeeded is a place holder to indicate that the schema version does not need to be specified.
+	VersionNotNeeded float32 = -18.0
+	// MinComponentYAMLVersion is the minimum schema version for the component
+	// YAML supported by this masonry toolchain.
+	MinComponentYAMLVersion float32 = 3.0
+	// MaxComponentYAMLVersion is the minimum schema version for the component
+	// YAML supported by this masonry toolchain.
+	MaxComponentYAMLVersion float32 = VersionNotNeeded
+)
```

Comment:
i agree, this isn't the best way period.

the opencontrol.yaml actually uses semver (and there's a nice library to handle this). i'm going to see if i can convert component.yaml to using semver without breaking all pre `v3.0` schemas.